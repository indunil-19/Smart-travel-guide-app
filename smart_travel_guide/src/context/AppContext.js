import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}