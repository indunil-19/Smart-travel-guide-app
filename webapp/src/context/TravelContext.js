import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 