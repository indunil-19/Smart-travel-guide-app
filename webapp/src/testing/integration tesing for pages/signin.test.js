import React from "react"
import {render,fireEvent,screen, cleanup,waitFor} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import SignIn from "../../screens/TravelPlanApp/signIn"
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import { createMemoryHistory } from 'history'

describe('signin page testing', () => {
    const server=setupServer(
        rest.post("/signin",(req,res,ctx)=>{
            return res(
                ctx.json({
                    message:"sign in successs",
                    user:{
                        _id:"5654545454545454"
                    }
                })
            )
        })
    )
    
    beforeEach(()=>{      
    })
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());
    afterEach(cleanup)

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><SignIn  /></TravelProvider></MemoryRouter>,div)
    })
    it("should login correctly if no error",async()=>{
        const history = createMemoryHistory();
        render(<MemoryRouter><Router history={history}><TravelProvider><SignIn/></TravelProvider></Router></MemoryRouter>)        
        fireEvent.change(screen.getByPlaceholderText("email"), {target:{value:"udayangana98@gmail.com"}})
        fireEvent.change(screen.getByPlaceholderText("password"), {target:{value:"1234"}})
        fireEvent.click(screen.getByTestId("signIn"))
        await waitFor(()=>{
            const toasts=screen.getAllByRole("list")
            expect(toasts[2]).toBeInTheDocument();
            expect(history.location.pathname).toBe('/travelPlan')
        })
        
    })

    it("shoud toast if error in login",async()=>{
        const server=setupServer(
            rest.post("/signin",(req,res,ctx)=>{
                if(!req.body.email || !req.body.password){
                    return res(
                        ctx.json({
                            error:"sign in error",
                        })
                    )}
                return res(
                    ctx.json({
                        error:"sign in error",
                    })
                )
            })
        )
        render(<MemoryRouter><TravelProvider><SignIn/></TravelProvider></MemoryRouter>)        
        fireEvent.change(screen.getByPlaceholderText("email"), {target:{value:"udayangana98@gmail.com"}})
        fireEvent.change(screen.getByPlaceholderText("password"), {target:{value:""}})
        fireEvent.click(screen.getByTestId("signIn"))
        await waitFor(()=>{
            const toasts=screen.getAllByRole("list")
            expect(toasts[2]).toBeInTheDocument();
            
        })
    })

    it("shoud toast if email is wrong",async()=>{
        
        render(<MemoryRouter><TravelProvider><SignIn/></TravelProvider></MemoryRouter>)        
        fireEvent.change(screen.getByPlaceholderText("email"), {target:{value:"wrongemail"}})
        fireEvent.change(screen.getByPlaceholderText("password"), {target:{value:""}})
        fireEvent.click(screen.getByTestId("signIn"))
        await waitFor(()=>{
            const toasts=screen.getAllByRole("list")
            expect(toasts[2]).toBeInTheDocument();
            
        })
    })

    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><SignIn/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

})
