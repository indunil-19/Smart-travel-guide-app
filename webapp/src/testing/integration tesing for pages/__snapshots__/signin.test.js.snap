// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signin page testing should match snapshot 1`] = `
<div
  className="css-1g0sdft"
>
  <div
    className="css-1tu6eaj"
  >
    <div
      className="chakra-stack css-1xzycj5"
    >
      <h2
        className="chakra-heading css-kbn7if"
      >
        SignIn
      </h2>
      <div
        className="chakra-input__group css-4302v8"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,8c-1.178,0-2,0.822-2,2s0.822,2,2,2s2-0.822,2-2S13.178,8,12,8z"
            fill="none"
          />
          <path
            d="M12,4c-4.337,0-8,3.663-8,8c0,2.176,0.923,4.182,2.39,5.641c0.757-1.8,2.538-3.068,4.61-3.068h2 c2.072,0,3.854,1.269,4.61,3.068C19.077,16.182,20,14.176,20,12C20,7.663,16.337,4,12,4z M12,14c-2.28,0-4-1.72-4-4s1.72-4,4-4 s4,1.72,4,4S14.28,14,12,14z"
            fill="none"
          />
          <path
            d="M13,16.572h-2c-1.432,0-2.629,1.01-2.926,2.354C9.242,19.604,10.584,20,12,20s2.758-0.396,3.926-1.073 C15.629,17.582,14.432,16.572,13,16.572z"
            fill="none"
          />
          <path
            d="M12,2C6.579,2,2,6.579,2,12c0,3.189,1.592,6.078,4,7.924V20h0.102C7.77,21.245,9.813,22,12,22s4.23-0.755,5.898-2H18 v-0.076c2.408-1.846,4-4.734,4-7.924C22,6.579,17.421,2,12,2z M8.074,18.927c0.297-1.345,1.494-2.354,2.926-2.354h2 c1.432,0,2.629,1.01,2.926,2.354C14.758,19.604,13.416,20,12,20S9.242,19.604,8.074,18.927z M17.61,17.641 c-0.757-1.8-2.538-3.068-4.61-3.068h-2c-2.072,0-3.854,1.269-4.61,3.068C4.923,16.182,4,14.176,4,12c0-4.337,3.663-8,8-8 s8,3.663,8,8C20,14.176,19.077,16.182,17.61,17.641z"
          />
          <path
            d="M12,6c-2.28,0-4,1.72-4,4s1.72,4,4,4s4-1.72,4-4S14.28,6,12,6z M12,12c-1.178,0-2-0.822-2-2s0.822-2,2-2s2,0.822,2,2 S13.178,12,12,12z"
          />
        </svg>
        <input
          className="chakra-input css-0"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="email"
          type="email"
          value=""
        />
      </div>
      <div
        className="chakra-input__group css-4302v8"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M18 8h2a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V9a1 1 0 0 1 1-1h2V7a6 6 0 1 1 12 0v1zM5 10v10h14V10H5zm6 4h2v2h-2v-2zm-4 0h2v2H7v-2zm8 0h2v2h-2v-2zm1-6V7a4 4 0 1 0-8 0v1h8z"
            />
          </g>
        </svg>
        <input
          className="chakra-input css-0"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
      </div>
      <button
        className="chakra-button css-5c2176"
        onClick={[Function]}
        type="button"
      >
        Sign In
      </button>
      <p
        className="chakra-text css-0"
      >
        Forget Password?
        <a
          href="/travelPlan/reset"
          onClick={[Function]}
        >
          Reset
        </a>
      </p>
      <p
        className="chakra-text css-0"
      >
        you don't have an account?
        <a
          href="/travelPlan/signup"
          onClick={[Function]}
        >
          SignUp
        </a>
      </p>
    </div>
  </div>
</div>
`;
